### PR TITLE
Spark Coverage

### PR DESCRIPTION
**Pull request by shehabgamin**

### CODE REVIEW

File: crates/sail-plan/src/function/scalar/datetime.rs [hunk 1]
```diff
@@ -1,46 +1,176 @@
+use datafusion::arrow::datatypes::{
```
Comment (shehabgamin) [inline crates/sail-plan/src/function/scalar/datetime.rs:1]: Code in this file will be refactored in my next pr.

File: crates/sail-plan/src/extension/function/size.rs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+};
+
+use crate::extension::function::functions_nested_utils::{
+    compute_array_dims, make_scalar_function,
+};
+
+// expr_fn::cardinality doesn't fully match expected behavior.
+// Spark's cardinality function seems to be the same as the size function.
+// `cardinality`: https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.functions.cardinality.html
+// `size`: https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.functions.size.html
```
Comment (linhr) [inline crates/sail-plan/src/extension/function/size.rs:20]: I've added back a deleted comment here for better context. Let me know if it is still relevant.

File: crates/sail-sql/src/utils.rs [hunk 3]
```diff
@@ -13,14 +14,24 @@ use crate::expression::from_ast_expression;
 /// retain their case sensitivity and must be preserved exactly as written.
 /// TODO: Make sure this gets called everywhere in sail-sql
 ///     because sail-plan spec expects the raw identifier.
-pub fn normalize_ident(id: ast::Ident) -> String {
-    let ast::Ident { value, quote_style } = id;
+pub fn normalize_ident(id: &ast::Ident) -> String {
+    let ast::Ident { value, quote_style } = id.to_owned();
```
Comment (linhr) [inline crates/sail-sql/src/utils.rs:18]: This is a nice change on the function interface!

**Review by linhr — APPROVED** [review]
Great work!

I made some small edits and added a few comments for future improvement (SQL processor, math functions, etc.). Overall the PR looks quite solid. It is exciting to see so many new tests passing in this PR! 🚀

**Pull request merged**